### ISSUE TITLE
dynamic breakpoint support

### ISSUE THREAD

**Issue by GentleCP** [ioncodes/idacode#41]
Hi, the recent version of idacode does not support adding breakpoints with vscode (F9), we have to add breakpoint() function in advance, which makes the debug process more complex. In previous versions of idacode, I remember it is feasible to add a breakpoint with vscode (F9) as long as one breakpoint() function has been added. Is it a bug that should be fixed or not? If not, is it possible to support adding a breakpoint with vscode? 

![image](https://github.com/ioncodes/idacode/assets/38175267/0828d695-efd4-465d-986f-cda7b0eecde6)

**Comment by ioncodes** [comment]
You're right that one `breakpoint()` used to be enough to make it work. This seems to be odd, but I actually have not tested idacode with the latest IDA. Which version are you on currently? Perhaps there's some sort of incompatibility issue. If you're using an older IDA version I'd probably need more information.

**Comment by GentleCP** [comment]
The following text includes the environment I have used:
- OS: ubuntu22.04
- IDA Pro: linux 7.5
- idacode: latest
- debugpy: 1.6.7.post1
- tornado: 6.3.3
- vscode idacode plugin: v0.3.0

**Comment by ioncodes** [comment]
Hmm, the IDA version should be supported, however, it might be a Linux issue. I don't think we ever tested idacode on Linux and macOS. I can only come up with it being either a platform issue or that there's a quirk in the Linux version of IDA. I don't have a Linux copy of IDA so I can't really reproduce it on my side. Do you have a VM with the same issue that you might be able to share with me (you can remove the ida.key if that's an issue)?

Alternative idea that I came up with while writing this issue: There used to be a debugging bug with specific debugpy versions. I know that `debugpy===1.0.0rc2` should have been stable at one point, can you try again with this specific version installed?

**Comment by GentleCP** [comment]
Sorry for forgetting to reply, well, I have tried debugpy===1.0.0rc2, however, when I connect ida with idacode, a new exception occurred as follows:
![image](https://github.com/ioncodes/idacode/assets/38175267/18e0aba5-d1cc-4408-98a8-2bc4cc10a271)

**Comment by GentleCP** [comment]
I also have tried some old version of idacode (0.2.2) and debugpy (1.5.0, 1.6.5,1.2.0,1.1.0), unfortunately, none of them could solve the above issue.

**Comment by KcHNST** [comment]
Instead of puting a hard `breakpoint()` I've had success using this pattern to force debugpy to start tracing on the current thread.
```python3
import debugpy

if debugpy.is_client_connected():
    debugpy.debug_this_thread()

# normal breakpoint can now be hit from this point onwards
```
